beluga/include/beluga/random/multivariate_distribution_traits.hpp

```diff
@@ -110,6 +112,58 @@ struct multivariate_distribution_traits<T, std::enable_if_t<std::is_base_of_v<So
   }
 };
 
+/// Specialization for types derived from Sophus::SO3Base.
+template <class T>
+struct multivariate_distribution_traits<T, std::enable_if_t<std::is_base_of_v<Sophus::SO3Base<T>, T>>> {
+  /// The scalar type.
+  using scalar_type = typename T::Scalar;
+
+  /// The result type representation.
+  using result_type = Sophus::SO3<scalar_type>;
+
+  /// The vector type.
+  using vector_type = typename Eigen::Matrix<scalar_type, 3, 1>;
+
+  /// The covariance matrix type.
+  using covariance_type = typename Eigen::Matrix<scalar_type, 3, 3>;
+
+  /// Convert from result to vector representation.
+  [[nodiscard]] static constexpr vector_type to_vector(const result_type& t) { return t.log(); }
+
+  /// Convert from vector to result representation.
+  [[nodiscard]] static constexpr result_type from_vector(const vector_type& v) {
+    return Sophus::SO3<scalar_type>::exp(v);
```

Comment:
nit: Maybe we can add a block comment with a summary of the PR notes here for posterity.

Reply:
Something along this lines?

```
    // Projecting a Gaussian through the exponential map does not yield a unimodal distribution.
    // Rotations in Lie groups don't have infinite support, making the term 'normal' distribution incorrect.
    // This approach follows the SO2/SE2 method, facing similar issues in 3D.
    // For small covariances, the resulting rotations in SO3 approximate a normal distribution.
```

Reply:
Done!